module/Photo/src/Photo/Controller/PhotoController.php

```diff
@@ -10,17 +10,21 @@ class PhotoController extends AbstractActionController
 
     public function indexAction()
     {
-        $albums = $this->getAlbumService()->getAlbums();
         //add any other special behavior which is required for the main photo page here later
-
+        $years = $this->getAlbumService()->getAlbumYears();
+        $year = $this->params()->fromRoute('year');
+        // If no year is supplied, use the current year.
+        if(is_null($year)) $year = (date("m") < 7) ? date("Y") - 1 : date("Y");
```

Comment:
Er moet altijd accolades na een if statement. Daarbij mag hier wel een beetje commentaar bij waar de 7 vandaan komt.

Daarbij lijkt het mij handiger om hem op 8 te zetten. Nu zouden de fotos van een constiborrel onder een vorig jaar vallen omdat per 1-7 het nieuwe schooljaar ingaat. Magic numbers ook als constantes definieren

Daarbij date("n") gebruiken. Er zit hier potentieel een heel nasty bug in, waardoor juli onder dit jaar valt, maar augustus en september onder het volgende schooljaar. date("m") geeft voor augustus 08 terug. Het nummer 08 word door de 0 als octaal geinterpreteerd wat dan word gezien als 0 (want 08 is geen geldig octaal number). Nu lijkt het na trial and error dat de string naar integer conversie altijd base 10 gebruikt, maar dat het dan zou werken is meer door de interne inconsistentie van php dan door goed doordachtheid.